test/server/core/card/Upgrade.spec.js

```diff
@@ -105,5 +105,33 @@ describe('Upgrade cards', function() {
                 expect(this.tielnFighter).toBeInLocation('discard');
             });
         });
+
+        describe('When an upgrade is attached', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [{ card: 'first-legion-snowtrooper', upgrades: ['experience'] }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'death-trooper', damage: 1 }],
+                        base: { card: 'dagobah-swamp', damage: 5 }
+                    }
+                });
+            });
+
+            it('First legion Snowtrooper attacking Death trooper should receive +2/+0, overwhelm and +1/+1 from experience upgrade. Defeating Death trooper and dealing 3 to base', function () {
```

Comment:
Missed this test case name, still needs to be updated. Thanks!

Reply:
updated the case name.